Dev

Stellareign opened the pull request:
Пересобранный проект

Stellareign committed:
Пересобранный проект

Stellareign committed:
Пересобранный проект

Stellareign committed:
Пересобранный проект

Stellareign committed:
Пересобранный проект

Stellareign committed:
переместила интефейсы в пакет
добавмила репозиторий юзера
добавила сущность картинки
Создала поля поля сущности пароля (он нужен отдельно???)

dmitry-bizin commented on pom.xml:
```diff
@@ -57,11 +58,51 @@
+    </dependency>
+    <dependency>
+      <groupId>org.modelmapper</groupId>
+      <artifactId>modelmapper</artifactId>
```
зачем вам 2 РАЗНЫЕ библиотеки для маппинга? надо выбрать одну и использовать её

dmitry-bizin commented on pom.xml:
```diff
@@ -57,11 +58,51 @@
+<!--SWAGGER-->
+    <dependency>
+      <groupId>io.springfox</groupId>
+      <artifactId>springfox-swagger2</artifactId>
```
зачем второй раз подключать?

dmitry-bizin commented on pom.xml:
```diff
@@ -57,11 +58,51 @@
+    </dependency>
+    <dependency>
+      <groupId>org.springdoc</groupId>
+      <artifactId>springdoc-openapi-ui</artifactId>
```
зачем вам 2 РАЗНЫЕ библиотеки для swgger ui? надо выбрать одну и использовать её

dmitry-bizin commented on pom.xml:
```diff
@@ -57,11 +58,51 @@
+    <!-- https://mvnrepository.com/artifact/commons-io/commons-io -->
+    <dependency>
+      <groupId>commons-io</groupId>
+      <artifactId>commons-io</artifactId>
```
не думаю, что это нужно, а на будущее подключать точно не стоит ничего, чтобы не засорять проект

dmitry-bizin commented on pom.xml:
```diff
@@ -57,11 +58,51 @@
+    <!-- https://mvnrepository.com/artifact/org.apache.commons/commons-lang3 -->
+    <dependency>
+      <groupId>org.apache.commons</groupId>
+      <artifactId>commons-lang3</artifactId>
```
не думаю, что это нужно, а на будущее подключать точно не стоит ничего, чтобы не засорять проект

Stellareign committed:
переместила интефейсы в пакет
добавила репозиторий юзера
добавила сущность картинки
Создала поля поля сущности пароля (он нужен отдельно???)
Перенесла маппер в Application.clss
создала неполноценный маппер на юзера (рабочий ???)

Stellareign committed:
переместила интефейсы в пакет
добавила репозиторий юзера
добавила сущность картинки
Создала поля поля сущности пароля (он нужен отдельно???)
Перенесла маппер в Application.clss
создала неполноценный маппер на юзера (рабочий ???)

dmitry-bizin commented on src/main/java/ru/skypro/homework/config/MapperUtil.java:
```diff
@@ -0,0 +1,20 @@
+    public ModelMapper getMapper() {
+        return new ModelMapper();
+    }
+    public static <R, E> List<R> convertList(List<E> list, Function<E, R> converter) {
```
почему метод static? надо убрать static, а вызывать через бин MapperUtil, который надо инжектить

dmitry-bizin commented on src/main/java/ru/skypro/homework/config/WebSecurityConfig.java:
```diff
@@ -58,5 +60,9 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
         return new BCryptPasswordEncoder();
     }
+    @Bean
+    public ModelMapper getMapper() {
```
зачем второй такой же бин? у вас уже объявлен в MapperUtil такой же

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,103 @@
+
+    private AdsService adsService;
+
+    private ModelMapper modelMapper;
```
мапперов здесь быть не должно, вся логика маппинга должна происходить в сервисах, чтобы сущности не пролезали в слой контроллеров

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,103 @@
+//                                           String nutrition, String health, String behavior) {
+//        return ownerReportService.editOwnerReportByIdFromController(ownerReportId, photo, nutrition, health, behavior);
+//    }
+    public ResponseEntity<CreateOrUpdateAdDTO> updateAd(int adId, @RequestParam String title, int price,
+                                                        String description) {
```
здесь неверно - должно быть тело запроса

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,103 @@
+            return new ResponseEntity<>(("Ошибка при попытке удалить Объявление с id = " + adId), HttpStatus.BAD_REQUEST);
+        }
+    }
+
```
не все методы из спецификации перенесены

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -0,0 +1,62 @@
+
+    @Operation(summary = "Получение списка всех комментариев")
+    @GetMapping
+    public ResponseEntity<CommentsDTO> getComment(int adsId) {
```
где аннотация? где параметр в пути?

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -0,0 +1,62 @@
+    // удаление комментария по id
+    @Operation(summary = "Удаление комментария")
+    @DeleteMapping("/ads/{adId}/comments/{commentId}")
+    public void deleteComment(@RequestParam int id) {
```
аннотация неправильная, параметров 2 в запросе, а не 1

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -0,0 +1,62 @@
+// (http://localhost:3000), даже если он отличается от домена, на котором запущено приложение.
+@RequiredArgsConstructor // генерирует конструктор с аргументами для всех полей, помеченных аннотацией @NonNull
+@RestController
+@RequestMapping("/comments")
```
базовый путь неправильный

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentsController.java:
```diff
@@ -0,0 +1,62 @@
+    // обновление комментария
+    @Operation(summary = "Обновление комментария")
+    @PatchMapping
+    public ResponseEntity<?> updateComment(@RequestParam int adId, int commentId, String text) {
```
здесь неверно - должно быть тело запроса

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/UserController.java:
```diff
@@ -0,0 +1,82 @@
+// (http://localhost:3000), даже если он отличается от домена, на котором запущено приложение.
+@RequiredArgsConstructor // генерирует конструктор с аргументами для всех полей, помеченных аннотацией @NonNull
+@RestController
+@RequestMapping("/user")
```
путь неверный

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/UserController.java:
```diff
@@ -0,0 +1,82 @@
+public class UserController {
+
+    private final AuthService authService;
+    AddUserDTO userDTO = new AddUserDTO();
```
зачем это?

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/UserController.java:
```diff
@@ -0,0 +1,82 @@
+    AddUserDTO userDTO = new AddUserDTO();
+
+    @Operation(summary = "Обновление пароля")
+    @PostMapping("/set_password/")
```
слэш на конце не нужен

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/UserController.java:
```diff
@@ -0,0 +1,82 @@
+
+    @Operation(summary = "Обновление пароля")
+    @PostMapping("/set_password/")
+    public ResponseEntity<PasswordDTO> setPassword(@RequestParam String currentPassword,
+                                                   @RequestParam String newPassword,
+                                                   Password password) {
```
параметры должны передаваться в теле запроса

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/UserController.java:
```diff
@@ -0,0 +1,82 @@
+    }
+
+    @Operation(summary = "Получение информации об авторизованном пользователе")
+    @GetMapping("/me/{email}")
```
такого эндпоинта нет в спецификации

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/UserController.java:
```diff
@@ -0,0 +1,82 @@
+
+    @Operation(summary = "Обновление информации об авторизованном пользователе")
+    @PatchMapping("/me")
+    public ResponseEntity<?> updateUser(String firstName, String lastName, String phone) {
```
данные передаются в теле запроса

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/UserController.java:
```diff
@@ -0,0 +1,82 @@
+    }
+
+    @Operation(summary = "Обновление аватара авторизованного пользователя")
+    @PatchMapping("/me/image")
+    public ResponseEntity<?> updateUserImage(String image) {
```
здесь неверно - запрос мультипартовый, также где аннотация? где сама картинка? она не в виде строки передаётся

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/ads/AdsDTO.java:
```diff
@@ -0,0 +1,23 @@
+@AllArgsConstructor
+public class AdsDTO {
+    int pk;
+    String imageAd;
```
поля неверные, не как в спецификации. Также все поля должны быть приватные

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/comments/CommentsDTO.java:
```diff
@@ -0,0 +1,31 @@
+
+    private int count; // общее количество комментариев
+
+    private List<Comments> allComm;
```
название поля неверное (см. спецификацию)

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/comments/NewCommentsDTO.java:
```diff
@@ -0,0 +1,27 @@
+@Data
+@AllArgsConstructor
+public class NewCommentsDTO {
+    private final int count; // общее количество комментариев
```
такого поля нет в DTO (см. спецификацию внимательнее)

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/comments/UpdateCommentsDTO.java:
```diff
@@ -0,0 +1,10 @@
+@Data
+public class UpdateCommentsDTO {
+    int adId;
+    int commentId;
```
поля неверные (см. спецификацию), также где private?

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/user/AddUserDTO.java:
```diff
@@ -0,0 +1,22 @@
+    private int id;
+    private String firstName;
+    private String lastName;
+    private String eMail;
```
поле неверное

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/user/UpdateUserDTO.java:
```diff
@@ -0,0 +1,10 @@
+@Data
+public class UpdateUserDTO {
+    private String firstName;
+    private String lastName;
```
поля неверные (см. спецификацию)

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/user/UpdateUserImageDTO.java:
```diff
@@ -0,0 +1,8 @@
+import lombok.Data;
+
+@Data
+public class UpdateUserImageDTO {
```
такого dto нет

dmitry-bizin commented on src/main/java/ru/skypro/homework/models/Ad.java:
```diff
@@ -0,0 +1,31 @@
+    private int price;              // 'цена объявления'
+    private String title;           // 'заголовок объявления'
+    private String adsDescription;  // 'описание объявления'
+    private int authorId;              // id автора объявления
```
здесь должна быть связь с юзером

dmitry-bizin commented on src/main/java/ru/skypro/homework/models/Comments.java:
```diff
@@ -0,0 +1,35 @@
+    private int pk;    // id комментания
+
+    @Column(name = "author_id",  nullable = false)
+    private int author; // id автора комментария
```
здесь должна быть связь с юзером

dmitry-bizin commented on src/main/java/ru/skypro/homework/models/Comments.java:
```diff
@@ -0,0 +1,35 @@
+    private long createdAt; // дата и время создания комментария в миллисекундах с 00:00:00 01.01.1970
+
+    @Column(name = "ads_id",  nullable = false)
+    private int adId;  // id объявления
```
здесь должна быть связь с объявлением

dmitry-bizin commented on src/main/java/ru/skypro/homework/models/Password.java:
```diff
@@ -0,0 +1,12 @@
+@Getter
+@Setter
+@EqualsAndHashCode
+public class Password {
```
зачем этот класс?

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/AdsServiceImpl.java:
```diff
@@ -0,0 +1,86 @@
+            throwException(String.valueOf(id));
+
+        }
+        Ad existingAd = (Ad) optionalAd.get();
```
зачем здесь каст?

dmitry-bizin commented on src/main/resources/application.properties:
```diff
@@ -0,0 +1,10 @@
+spring.datasource.url=jdbc:postgresql://localhost:5432/diplom
+spring.datasource.username=postgres
+spring.datasource.password=1234
+spring.jpa.hibernate.ddl-auto=update
```
```suggestion
spring.jpa.hibernate.ddl-auto=validate
```
при использовании liquibase стратегия должна быть validate, иначе смысла в liquibase нет никакого

dmitry-bizin commented on src/main/java/ru/skypro/homework/models/Comments.java:
```diff
@@ -0,0 +1,35 @@
+/*
+проработать связь сущностей автор-комментарий и/или комментарий-объявление
+ */
+@Component
```
зачем здесь эта аннотация? она не нужна

dmitry-bizin commented on src/main/java/ru/skypro/homework/models/Ad.java:
```diff
@@ -0,0 +1,31 @@
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
+@Component
```
зачем здесь эта аннотация? она не нужна

dmitry-bizin commented on src/main/java/ru/skypro/homework/models/User.java:
```diff
@@ -0,0 +1,41 @@
+
+@Data
+@Entity
+@Component
```
зачем здесь эта аннотация? она не нужна

alkor63 committed:
Дописал все методы в контроллере+сервисе объявлений

alkor63 committed:
Дописал все методы в контроллере+сервисе объявлений

alkor63 committed:
Дописал все методы в контроллере+сервисе объявлений

alkor63 committed:
Дописал все методы в контроллере+сервисе объявлений

alkor63 committed:
Внёс пару изменений, теперь даже компилируется ;-)

alkor63 committed:
Внёс пару изменений, теперь даже компилируется ;-)

alkor63 committed:
Внёс пару изменений, теперь даже компилируется ;-)

alkor63 committed:
Внёс пару изменений, теперь даже компилируется ;-)

alkor63 committed:
Внёс пару изменений, теперь даже компилируется ;-)

alkor63 committed:
Внёс пару изменений, теперь даже компилируется ;-)

Stellareign committed:
Merge remote-tracking branch 'origin/RuAna' into RuAna

Stellareign committed:
переместила интефейсы в пакет
добавила репозиторий юзера
добавила сущность картинки
Создала поля поля сущности пароля (он нужен отдельно???)
Перенесла маппер в Application.clss
создала неполноценный маппер на юзера (рабочий ???)

Stellareign committed:
БД починились
Пробная регистрация в сваггере прошла

alkor63 committed:
повторная попытка пуша в dev

alkor63 committed:
повторная попытка пуша в dev

alkor63 committed:
Merge remote-tracking branch 'origin/Aleksey' into Aleksey

# Conflicts:
#	src/main/java/ru/skypro/homework/entity/Ad.java
#	src/main/java/ru/skypro/homework/service/impl/AdsServiceImpl.java
#	src/main/resources/application.properties

Rumonz1 committed:
создал несколько классов по комментариям

Rumonz1 committed:
создал несколько классов по комментариям

Rumonz1 committed:
Подправил названия

Rumonz1 committed:
Подправил названия

Stellareign committed:
БД снова запускаются
контроллеры не работают :(

alkor63 committed:
Внес исправления в маппер, контроллер-сервер объявлений
Пока не разобрался почему в БД объявления корректно добавляются, редактируются и удаляются, но в Сваггере "код ошибки 500"

Rumonz1 committed:
фиксирую малые изменения, пока без логики

Rumonz1 committed:
построил логику контроллеров(не до конца)

Rumonz1 committed:
Merge branch 'Timur' into dev

Stellareign committed:
Небольшой рефакторинг перед мержем и маленькая правка ексепшена в сервисе комментов

Stellareign committed:
Разрешен конфликт слияния

Stellareign committed:
Merge branch 'Aleksey' into dev
